llvm/lib/Transforms/Scalar/LICM.cpp

```diff
@@ -857,9 +857,19 @@ class ControlFlowHoister {
     }
 
     // Now finally clone BI.
-    ReplaceInstWithInst(
-        HoistTarget->getTerminator(),
-        BranchInst::Create(HoistTrueDest, HoistFalseDest, BI->getCondition()));
+    auto *NewBI =
+        BranchInst::Create(HoistTrueDest, HoistFalseDest, BI->getCondition(),
+                           HoistTarget->getTerminator()->getIterator());
+    HoistTarget->getTerminator()->eraseFromParent();
+    // Handle "performance-related" metadata. In particular:
```

Comment:
nit: Simplify the comment now that we only handle md prof in this patch? E.g. "Handle md_prof: it should also come ..."